File: apps/server/internal/todo/domain/todo.go
package domain

import "time"

type TodoList struct {
	ID                 string    `json:"id"`
	UserID             string    `json:"user_id"`
	GoogleID           *string   `json:"google_id,omitempty"`
	GoogleAccountID    *string   `json:"google_account_id,omitempty"`
	GoogleAccountEmail *string   `json:"google_account_email,omitempty"`
	Name               string    `json:"name"`
	SortOrder          int       `json:"sort_order"`
	ActiveCount        int       `json:"active_count"`
	DoneCount          int       `json:"done_count"`
	TotalCount         int       `json:"total_count"`
	Source             string    `json:"source"` // google | local
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`
}

type Todo struct {
	ID        string     `json:"id"`
	ListID    string     `json:"list_id"`
	UserID    string     `json:"user_id"`
	ParentID  *string    `json:"parent_id"`
	GoogleID  *string    `json:"google_id,omitempty"`
	Title     string     `json:"title"`
	Notes     string     `json:"notes"`
	Due       *string    `json:"due"` // YYYY-MM-DD
	Priority  string     `json:"priority"`
	IsDone    bool       `json:"is_done"`
	IsPinned  bool       `json:"is_pinned"`
	SortOrder int        `json:"sort_order"`
	DoneAt    *time.Time `json:"done_at,omitempty"`
	CreatedAt time.Time  `json:"created_at"`
	UpdatedAt time.Time  `json:"updated_at"`
	DeletedAt *time.Time `json:"deleted_at,omitempty"`
}

File: packages/features/todo/src/domain/Todo.ts
export interface Todo {
  id: string;
  listId: string;
  userId: string;
  parentId: string | null;
  title: string;
  notes: string;
  dueDate: string | null;
  dueTime: string | null;
  isDone: boolean;
  isPinned: boolean;
  starredAt?: string | null;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
}

export function normalizeTodoTitle(raw: string): string {
  const title = raw.trim();
  if (!title) {
    throw new Error("title is required");
  }
  return title;
}
